stl/inc/numeric

```diff
@@ -625,6 +625,20 @@ _NODISCARD constexpr common_type_t<_Mt, _Nt> gcd(const _Mt _Mx, const _Nt _Nx) n
     using _Common          = common_type_t<_Mt, _Nt>;
     using _Common_unsigned = make_unsigned_t<_Common>;
 
+#if _HAS_CXX20

```

Comment:
I've come up with some (IMO) simple works that backports the checks to C++17. Not sure whether this should be done in a follow-up PR.

Reply:
I'm not sure if lowering checked multiplication to pre-concepts is a good idea, but if it is, 6ou can do a follow-up or even as a PR to my PR

Reply:
Opened AlexGuteniev/STL#6.

Reply:
Thanks, looks good!

Reply:
It is now easy to avoid `if constexpr (true)` while still avoiding complex preprocessing, I'll do a follow up for that